sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -6305,39 +6298,43 @@ pi_result piEnqueueMemBufferMap(pi_queue Queue, pi_mem Buffer,
   // Lock automatically releases when this goes out of scope.
   std::scoped_lock Lock(Queue->Mutex, Buffer->Mutex);
 
-  // For discrete devices we need a command list
-  pi_command_list_ptr_t CommandList{};
-  if (auto Res = Queue->Context->getAvailableCommandList(Queue, CommandList))
-    return Res;
-
-  // Set the commandlist in the event
-  if (Event) {
-    (*Event)->ZeCommandList = CommandList->first;
-    CommandList->second.append(*Event);
-    PI_CALL(piEventRetain(*Event));
-  }
-
   if (Buffer->MapHostPtr) {
     *RetMap = Buffer->MapHostPtr + Offset;
   } else {
+    // TODO: use USM host allocator here
     if (auto Res = ZeHostMemAllocHelper(RetMap, Queue->Context, Size))
```

Comment:
Can we replace this call with piextUSMHostAlloc right away? (considering that you are removing the switch "enableBufferPooling()" which controls pooling)
```suggestion
    if (auto Res = piextUSMHostAlloc(RetMap, Queue->Context, nullptr, Size, 1))
```

Reply:
I retained enableBufferPooling